tools/metadata/index.ts

```diff
@@ -4,18 +4,34 @@ import { TokenInfo } from '@uniswap/token-lists'
 import { defaults, known } from './src/constants'
 import { getTokenName, getTokenSymbol, getTokenDecimals } from './src/helpers'
 
-export async function fetchTokens(chainId: number): Promise<Array<TokenInfo>> {
-  const tokens = await Promise.all(
+export async function fetchTokens(
+  chainId: number
+): Promise<{ tokens: TokenInfo[]; errors: string[] }> {
+  const errors = []
+  let tokens = []
+  const promises = await Promise.allSettled(
     known.map(async url => {
-      const { data } = await axios.get(url)
-      return data.tokens
+      try {
+        const res = await axios.get(url)
+        return res.data.tokens
+      } catch (e) {
+        throw new Error(e.message)
```

Comment:
Is it necessary to throw a new Error with the same message?

Reply:
Nope, it's unnecessary, just pushed up a commit removing it.

Reply:
Actually meant to ask whether the try / catch here is necessary. If `axios.get` throws will the encapsulating promise status not be `rejected`?

Reply:
Ahh didn't realize the error bubbles to the encapsulating promise, okay pushing up a commit removing the try / catch